### PR TITLE
[embree3] Update to 3.11.0

### PR DESCRIPTION
**Pull request by regen100**
Update to the latest release v3.11.0.
Outdated patches are replaced.

### CODE REVIEW

File: ports/embree3/portfile.cmake [hunk 1]
```diff
@@ -1,64 +1,40 @@
...
 
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/share)
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
 if(VCPKG_LIBRARY_LINKAGE STREQUAL static)
-  file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/bin ${CURRENT_PACKAGES_DIR}/debug/bin)
+    file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/bin ${CURRENT_PACKAGES_DIR}/debug/bin)
 endif()
-if (EXISTS ${CURRENT_PACKAGES_DIR}/uninstall.command)
-    file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/uninstall.command)
+if(APPLE)
```
Comment (NancyLi1013) [inline ports/embree3/portfile.cmake:34]: Is this command only required on non-osx platform?
if so, please use `VCPKG_TARGET_IS_OSX `to judge it.

File: ports/embree3/portfile.cmake [hunk 2]
```diff
@@ -1,64 +1,40 @@
...
+    file(REMOVE ${CURRENT_PACKAGES_DIR}/uninstall.command ${CURRENT_PACKAGES_DIR}/debug/uninstall.command)
 endif()
-if (EXISTS ${CURRENT_PACKAGES_DIR}/debug/uninstall.command)
-    file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/uninstall.command)
-endif()
-
 file(RENAME ${CURRENT_PACKAGES_DIR}/share/doc ${CURRENT_PACKAGES_DIR}/share/embree/doc)
```
Comment (NancyLi1013) [inline ports/embree3/portfile.cmake:37]:
```suggestion
file(RENAME ${CURRENT_PACKAGES_DIR}/share/doc ${CURRENT_PACKAGES_DIR}/share/${PORT}/doc)
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/embree3/portfile.cmake (hunk 1 above)
Comment (regen100) [reply ports/embree3/portfile.cmake:34]: embree3 installs the file unnecessary for vcpkg on `APPLE`
https://github.com/embree/embree/blob/cc8c3aac00cdb63fb1acb44bcd0fa89a17139533/common/cmake/package.cmake#L91-L94

**Review by NancyLi1013 — APPROVED** [review]

**Review by NancyLi1013 — APPROVED** [review]

File: ports/embree3/usage [hunk 3]
```diff
@@ -0,0 +1,4 @@
+The package embree3 provides CMake targets:
+
+    find_package(embree 3 CONFIG REQUIRED)
```
Comment (strega-nil) [inline ports/embree3/usage:3]:
```suggestion
    find_package(embree3 CONFIG REQUIRED)
```

Isn't this the correct `find_package`?

Comment (regen100) [reply ports/embree3/usage:3]: Yes. Related to #8608, users often use `find_package(embree 3)`.
And also, [Upstream README](https://github.com/embree/embree#using-embree) says using `FIND_PACKAGE(embree 3.0 REQUIRED)`.

Comment (strega-nil) [reply ports/embree3/usage:3]: Ah, alright, cool :)

**Pull request merged**